File: Pharmaceutical_Company_Example/02_Structural/PHP/Bridge.php
<?php

// Abstraction: RegulatoryProcess
class RegulatoryProcess {
    protected RegulatoryImplementation $implementation;

    public function __construct(RegulatoryImplementation $implementation) {
        $this->implementation = $implementation;
    }

    public function submit(): void {
        $this->implementation->submit();
    }

    public function approve(): void {
        $this->implementation->approve();
    }
}

// Implementation Interface
interface RegulatoryImplementation {
    public function submit(): void;
    public function approve(): void;
}

// Concrete Implementation for Vaccines
class VaccineProcess implements RegulatoryImplementation {
    public function submit(): void {
        echo "Submitting vaccine clinical trial data...\n";
    }

    public function approve(): void {
        echo "Vaccine regulatory approval granted.\n";
    }
}

// Concrete Implementation for Small-Molecule Drugs
class SmallMoleculeDrugProcess implements RegulatoryImplementation {
    public function submit(): void {
        echo "Submitting small-molecule clinical trial data...\n";
    }

    public function approve(): void {
        echo "Small-molecule regulatory approval granted.\n";
    }
}

// Concrete Implementation for Biologics
class BiologicsProcess implements RegulatoryImplementation {
    public function submit(): void {
        echo "Submitting biologics clinical trial data...\n";
    }

    public function approve(): void {
        echo "Biologics regulatory approval granted.\n";
    }
}

// Usage
$preclinicalSmallMolecule = new RegulatoryProcess(new SmallMoleculeDrugProcess());
$preclinicalSmallMolecule->submit();
$preclinicalSmallMolecule->approve();

$clinicalBiologics = new RegulatoryProcess(new BiologicsProcess());
$clinicalBiologics->submit();
$clinicalBiologics->approve();

$clinicalVaccine = new RegulatoryProcess(new VaccineProcess());
$clinicalVaccine->submit();
$clinicalVaccine->approve();

?>

File: Pharmaceutical_Company_Example/02_Structural/Java/Bridge.java
// Interface for the implementation
interface RegulatoryImplementation {
    void submit();
    void approve();
}

// Abstraction class
class RegulatoryProcess {
    private RegulatoryImplementation implementation;

    public RegulatoryProcess(RegulatoryImplementation implementation) {
        this.implementation = implementation;
    }

    public void submit() {
        implementation.submit();
    }

    public void approve() {
        implementation.approve();
    }
}

// Concrete implementation for Vaccines
class VaccineProcess implements RegulatoryImplementation {
    public void submit() {
        System.out.println("Submitting vaccine clinical trial data...");
    }

    public void approve() {
        System.out.println("Vaccine regulatory approval granted.");
    }
}

// Concrete implementation for Small-Molecule Drugs
class SmallMoleculeDrugProcess implements RegulatoryImplementation {
    public void submit() {
        System.out.println("Submitting small-molecule clinical trial data...");
    }

    public void approve() {
        System.out.println("Small-molecule regulatory approval granted.");
    }
}

// Concrete implementation for Biologics
class BiologicsProcess implements RegulatoryImplementation {
    public void submit() {
        System.out.println("Submitting biologics clinical trial data...");
    }

    public void approve() {
        System.out.println("Biologics regulatory approval granted.");
    }
}

// Main class to demonstrate usage
public class Main {
    public static void main(String[] args) {
        RegulatoryProcess preclinicalSmallMolecule = new RegulatoryProcess(new SmallMoleculeDrugProcess());
        preclinicalSmallMolecule.submit();
        preclinicalSmallMolecule.approve();

        RegulatoryProcess clinicalBiologics = new RegulatoryProcess(new BiologicsProcess());
        clinicalBiologics.submit();
        clinicalBiologics.approve();

        RegulatoryProcess clinicalVaccine = new RegulatoryProcess(new VaccineProcess());
        clinicalVaccine.submit();
        clinicalVaccine.approve();
    }
}
